src/components/member-role-update/index.js

```diff
@@ -75,30 +89,86 @@ const MemberRoleUpdate = () => {
 
   const memberRoleUpdateButton = (
     <button
+      data-testid="promoteDemoteButton"
       className={classNames.moveToMember}
       type="button"
       onClick={() => promoteDemoteAMember(userId)}
     >
       {member ? 'Demote Member' : 'Promote to Member'}
     </button>
   );
-
+  const handleValidReason = (id, reason) => {
+    const isEmptyReason = !reason.length; // check for empty string
+    const isReasonEmptyOrWhitespace = /^\s*$/.test(reason); // check for empty or multiple whitespaces
+    const isMoreThan99Words = reason.split(' ').length > 99; // check for more than 99 words
+    const isMoreThan50Characters = reason.length <= 25; // check for more than 25 characters
+    switch (!archived) {
+      case isEmptyReason:
+        setvalidateError('Reason cannot be empty!');
+        break;
+      case isReasonEmptyOrWhitespace:
+        setvalidateError('Reason cannot be empty or multiple whitespaces!');
+        break;
+      case isMoreThan99Words:
+        setvalidateError('Reason cannot be more than 99 words!');
+        break;
+      case isMoreThan50Characters:
+        setvalidateError('Reason should have more than 25 characters!');
+        break;
+      default:
+        setvalidateError('');
+    }
+    const isValid =
+      !isEmptyReason &&
+      !isReasonEmptyOrWhitespace &&
+      !isMoreThan99Words &&
+      !isMoreThan50Characters;
+    if (isValid || archived) {
+      archiveUnArchiveTheMember(id, reason); // if valid reason or unarchiving, then archive/unarchive
```

Comment:
Here also. There is no need to put comments for logic you implemented.

Reply:
updated